lib/engine/game/g_18_uruguay/game.rb

```diff
@@ -0,0 +1,908 @@
+              type: :destination_bonus,
+              description: destination_description(c),
+              count: 1
+            )
+            c.add_ability(ability)
+
+            dest_hex.tile.icons << Part::Icon.new("../#{c.destination_icon}", "#{c.id}_destination")
+          end
+        end
+
+        def destination_description(corporation)
+          dest_hex = hex_by_id(corporation.destination_coordinates)
+
+          "Destination bonus: #{dest_hex.location_name} (#{dest_hex.name})"
+        end
+
+        def corn_farm
+          @corn_farm ||= company_by_id('LA_CORN')
+        end
+
+        def sheep_farm
+          @sheep_farm ||= company_by_id('LO_SHEEP')
+        end
+
+        def cattle_farm
+          @cattle_farm ||= company_by_id('LO_CATTLE')
+        end
+
+        def train_with_goods?(train)
+          return unless train
+
+          train.name.include?(self.class::GOODS_TRAIN)
+        end
+
+        def stock_round
+          Round::Stock.new(self, [
+            Engine::Step::DiscardTrain,
+            Engine::Step::Exchange,
+            Engine::Step::SpecialTrack,
+            G18Uruguay::Step::BuySellParShares,
+          ])
+        end
+
+        def operating_round(round_num)
+          Round::Operating.new(self, [
+            Engine::Step::Bankrupt,
+            Engine::Step::Exchange,
+            G18Uruguay::Step::CornFarm,
+            G18Uruguay::Step::SheepFarm,
+            G18Uruguay::Step::CattleFarm,
+            Engine::Step::SpecialTrack,
+            Engine::Step::SpecialToken,
+            G18Uruguay::Step::TakeLoanBuyCompany,
+            Engine::Step::HomeToken,
+            G18Uruguay::Step::Track,
+            G18Uruguay::Step::Token,
+            G18Uruguay::Step::Route,
+            G18Uruguay::Step::Dividend,
+            G18Uruguay::Step::DiscardTrain,
+            G18Uruguay::Step::BuyTrain,
+            [G18Uruguay::Step::TakeLoanBuyCompany, { blocks: true }],
+          ], round_num: round_num)
+        end
+
+        def new_nationalization_round(round_num)
+          G18Uruguay::Round::Nationalization.new(self, [
+              G18Uruguay::Step::DiscardTrain,
+              G18Uruguay::Step::PayoffLoans,
+              G18Uruguay::Step::NationalizeCorporation,
+              G18Uruguay::Step::RemoveTokens,
+              ], round_num: round_num)
+        end
+
+        def multiple_buy_only_from_market?
+          !optional_rules&.include?(:multiple_brown_from_ipo)
+        end
+
+        def home_token_locations(corporation)
+          raise NotImplementedError unless corporation.name == 'FCE'
+
+          hexes = @hexes.dup
+          hexes.select do |hex|
+            hex.tile.cities.any? { |city| city.tokenable?(corporation, free: true) }
+          end
+        end
+
+        def reset_adjustable_trains!(_entity, routes)
+          routes.each do |route|
+            p 'Find a way to clear train from good' if train_with_goods?(route.train)
```

Comment:
Could this be commented out until implemented, to avoid accidental logging to the console?

```suggestion
            # p 'Find a way to clear train from good' if train_with_goods?(route.train)
```